build.gradle

```diff
@@ -259,6 +266,97 @@ testClusters.integTest {
     }
 }
 
+String bwcVersion = "1.13.0.0";
+String baseName = "adMixedCluster"
+String bwcFilePath = "src/test/resources/org/opensearch/ad/bwc/"
+
+testClusters {
+    "${baseName}" {
+        testDistribution = "ARCHIVE"
+        versions = ["7.10.2","1.0.0"]
+        numberOfNodes = 3
+        plugin(provider(new Callable<RegularFile>(){
+            @Override
+            RegularFile call() throws Exception {
+                return new RegularFile() {
+                    @Override
+                    File getAsFile() {
+                        return fileTree(bwcFilePath + "job-scheduler/" + bwcVersion).getSingleFile()
+                    }
+                }
+            }
+        }))
+        plugin(provider(new Callable<RegularFile>(){
+            @Override
+            RegularFile call() throws Exception {
+                return new RegularFile() {
+                    @Override
+                    File getAsFile() {
+                        return fileTree(bwcFilePath + "anomaly-detection/" + bwcVersion).getSingleFile()
+                    }
+                }
+            }
+        }))
+        setting 'repositories.url.allowed_urls', 'http://snapshot.test*'
+        setting 'path.repo', "${buildDir}/cluster/shared/repo/${baseName}"
+        setting 'http.content_type.required', 'true'
+    }
+}
+
+task "${baseName}#oldVersionClusterTask"(type: StandaloneRestIntegTestTask) {
+    useCluster testClusters."${baseName}"
+    if (System.getProperty("mixedCluster") != null) {
+        filter {
+            includeTestsMatching "org.opensearch.ad.bwc.*IT"
+        }
+    }
+    systemProperty 'tests.rest.suite', 'old_cluster'
+    systemProperty 'tests.plugin_bwc_version', bwcVersion
+    nonInputProperties.systemProperty('tests.rest.cluster', "${-> testClusters."${baseName}".allHttpSocketURI.join(",")}")
+    nonInputProperties.systemProperty('tests.clustername', "${-> testClusters."${baseName}".getName()}")
+}
+
+task "${baseName}#mixedClusterTask"(type: StandaloneRestIntegTestTask) {
```

Comment:
Couple of comments:
1. I see `build.gradle` is creeping up with lots of changes.
Can all of this logic be abstracted in `bwc.gradle` and consumed here? 
2. Can you add comments to each of the steps, it would be helpful for other plugins owners to understand why these steps are required.

Reply:
1. I have created an issue for this #167, will work on that.
2. Added comments in the steps for all types of upgrades.